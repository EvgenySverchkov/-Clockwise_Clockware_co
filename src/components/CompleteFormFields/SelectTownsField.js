import React from "react";

import FormGroup from "../FormComponents/FormGroup";
import CheckBtn from "../FormComponents/CheckBtn";

const SelectTownsField = ({ townsArr, changeHandler }) => (
  <FormGroup isRow={false}>
    <div className="mb-2 font-weight-bold">Choose towns</div>
    <div>
      {townsArr.length === 0 ? (
        <div>-- List is empty --</div>
      ) : (
        townsArr.map((item) => (
          <CheckBtn
            name={item.name}
            nm={"towns"}
            id={item.id}
            key={item.name}
            changeHandler={changeHandler}
          />
        ))
      )}
    </div>
  </FormGroup>
);

export default SelectTownsField;
