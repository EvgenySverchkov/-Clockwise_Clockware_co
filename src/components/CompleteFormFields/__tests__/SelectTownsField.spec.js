import React from "react";

import {create} from "react-test-renderer";
import SelectTownsField from "../SelectTownsField";

it("<SelectTownsField /> test which is used in forms", ()=>{
    const mockProps = {
        changeHandler: jest.fn(),
        townsArr: [{id: 1, name: "SomeTown"}, {id: 2, name: "SomeTown2"}]
    }
    const component = create(<SelectTownsField {...mockProps}/>);
    expect(component.toJSON()).toMatchSnapshot();
})