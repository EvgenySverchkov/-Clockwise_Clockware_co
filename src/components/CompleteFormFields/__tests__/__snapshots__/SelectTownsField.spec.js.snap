// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectTownsField /> test which is used in forms 1`] = `
<div
  className="form-group  text-center text-sm-left col-12 col-md-10 col-lg-8"
>
  <div
    className="mb-2 font-weight-bold"
  >
    Choose towns
  </div>
  <div>
    <div
      className="form-check-inline"
    >
      <label
        className="form-check-label"
        htmlFor="SomeTown"
      >
        <input
          className="form-check-input towns"
          id="SomeTown"
          name="towns"
          onChange={[MockFunction]}
          type="checkbox"
          value="SomeTown"
        />
        SomeTown
      </label>
    </div>
    <div
      className="form-check-inline"
    >
      <label
        className="form-check-label"
        htmlFor="SomeTown2"
      >
        <input
          className="form-check-input towns"
          id="SomeTown2"
          name="towns"
          onChange={[MockFunction]}
          type="checkbox"
          value="SomeTown2"
        />
        SomeTown2
      </label>
    </div>
  </div>
</div>
`;
